flow/layers/image_filter_layer.cc

```diff
@@ -39,12 +40,34 @@ void ImageFilterLayer::Diff(DiffContext* context, const Layer* old_layer) {
 void ImageFilterLayer::Preroll(PrerollContext* context,
                                const SkMatrix& matrix) {
   TRACE_EVENT0("flutter", "ImageFilterLayer::Preroll");
+
   Layer::AutoPrerollSaveLayerState save =
       Layer::AutoPrerollSaveLayerState::Create(context);
 
+  context->raster_cached_entries.emplace_back(
+      RasterCacheEntry(this, *context, matrix));
+  auto current_index = context->raster_cached_entries.size();
+
+  auto& cache_entry = context->raster_cached_entries.back();
+
   SkRect child_bounds = SkRect::MakeEmpty();
   PrerollChildren(context, matrix, &child_bounds);
 
+  cache_entry.num_child_entries =
+      context->raster_cached_entries.size() - current_index;
+  auto cache_type = NeedCaching(context, matrix);
+
+  cache_entry.need_caching = false;
+  if (cache_type == CacheableLayer::CacheType::kCurrent) {
+    cache_entry.layer = this;
```

Comment:
the layer pointer already points to this layer so this line does nothing.

Also, the kCurrent return value means we should cache this layer, but need_caching is false at this point and you don't update it to true.

Reply:
ha, this is my wrong, sorry, have fix it .done